src/Core/Form/IdentifiableObject/DataProvider/CarrierFormDataProvider.php

```diff
@@ -81,4 +94,83 @@ public function getDefaultData()
             ],
         ];
     }
+
+    /**
+     * Function to format ranges data.
+     *
+     * @param CarrierRangesCollection $carrierRangesCollection
+     *
+     * @return array
+     */
+    private function formatRangesData(CarrierRangesCollection $carrierRangesCollection): array
+    {
+        $ranges = [];
+
+        // For each zones, we need to get all ranges
+        foreach ($carrierRangesCollection->getZones() as $zone) {
+            foreach ($zone->getRanges() as $range) {
+                $ranges[] = [
+                    'from' => $range->getFrom()->__toString(),
+                    'to' => $range->getTo()->__toString(),
+                ];
+            }
+        }
+
+        // Then, we remove duplicates and sort ranges by from value.
+        $ranges = array_unique($ranges, SORT_REGULAR);
+        $from_values = array_column($ranges, 'from');
+        array_multisort($from_values, SORT_ASC, $ranges);
+
+        return ['data' => json_encode($ranges)];
+    }
+
+    /**
+     * Function to format ranges costs data.
+     *
+     * @param CarrierRangesCollection $carrierRangesCollection
+     *
+     * @return array
+     */
+    private function formatRangesCostsData(EditableCarrier $carrier, CarrierRangesCollection $carrierRangesCollection): array
+    {
+        $ranges = [];
+
+        // We retrieve zones to get the correct zone name
+        $zones = $this->zonesChoiceProvider->getChoices([]);
+        $zones = array_flip($zones);
+
+        // We choose the right symbol for the range in function of the ShippingMethod
+        switch ($carrier->getShippingMethod()) {
+            default:
+                $rangeSymbol = '';
+                break;
+            case ShippingMethod::BY_PRICE:
+                $rangeSymbol = $this->currencyDataProvider->getDefaultCurrencySymbol();
+                break;
+            case ShippingMethod::BY_WEIGHT:
+                $rangeSymbol = $this->configuration->get('PS_WEIGHT_UNIT');
+                break;
+        }
+
+        // For each zones, we need to get all ranges
+        foreach ($carrierRangesCollection->getZones() as $zone) {
+            $zoneRanges = [];
+            foreach ($zone->getRanges() as $range) {
+                $zoneRanges[] = [
+                    'range' => $range->getFrom()->__toString() . $rangeSymbol . ' - ' . $range->getTo()->__toString() . $rangeSymbol,
```

Comment:
The suggestion to improve the range label in JS is similar here, a mix between CLDR and trans would be nice as an improvement (not in this PR)